### PR TITLE
Updates ESLint to 8.0.0

### LINKED ISSUE #38: ESLint v8 Compatibility?
With `@typescript-eslint/*` now compatible with `v8`, this seems feasible now with `package.json` updates and this:

https://gist.github.com/lydell/d62ce96c95c035811133a5396195da14

PS: it would be nice to have this plugin formally included in the `eslint` rules instead of inheriting from `unused-vars`.

The `--fix` functionality included is very useful!

I imagine you already considered this but mentioning it just in case.

### PR DESCRIPTION
**Pull request by benkroeger**
fixes #38 

includes BREAKING CHANGES

- drops node 10 support
- drops eslint@6
- drops eslint@7
- drops @typescript-eslint/eslint-plugin@4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,17 +29,21 @@
 		"eslint-rule-composer": "^0.3.0"
 	},
 	"devDependencies": {
-		"@typescript-eslint/eslint-plugin": "^4.14.2",
-		"@typescript-eslint/parser": "^4.14.2",
-		"eslint": "^7.19.0",
-		"jest": "^27.0.6",
-		"typescript": "^4.0.3"
```
Comment (gfyoung) [inline package.json:36]: Was there ever any use for this package?

File: package.json [hunk 2]
```diff
@@ -29,17 +29,21 @@
...
 	},
 	"engines": {
-		"node": "^10.12.0 || >=12.0.0"
+		"node": "^12.22.0 || ^14.17.0 || >=16.0.0"
 	},
 	"license": "MIT",
 	"repository": "https://github.com/sweepline/eslint-plugin-unused-imports",
 	"homepage": "https://github.com/sweepline/eslint-plugin-unused-imports",
-	"bugs": "https://github.com/sweepline/eslint-plugin-unused-imports/issues/"
+	"bugs": "https://github.com/sweepline/eslint-plugin-unused-imports/issues/",
```
[the remark this answers is not in the document]
Comment (benkroeger) [reply package.json:43]: doesn't seem related to the topic of this PR. Do you mind fixing cosmetics separately?
If you insist, I'll do the change though

[the remark this answers is not in the document]
Comment (gfyoung) [reply package.json:43]: 🤷 - I agree with your comments, which is why it's a nit.

@sweepline: I'll defer to you whether you want to include this cosmetic change or consider an entirely separate PR.

**Comment by sweepline** [comment]
Sorry about the wait... I've published this as v2.0.0

**Pull request merged**